### ISSUE TITLE
SmallRye REST client - warnings about duplicate provider registration

### ISSUE THREAD

**Issue by mkouba** [quarkusio/quarkus#2647]
When I add the `quarkus-smallrye-rest-client` extension to my app I get a bunch of warnings:
```
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.IIOImageProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.sse.SseEventOutputProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.sse.SseEventProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.DefaultTextPlain is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.client.jaxrs.internal.CompletionStageRxInvokerProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.ReactiveStreamProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.interceptors.CacheControlFeature is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.interceptors.ServerContentEncodingAnnotationFeature is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.FormUrlEncodedProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.CompletionStageProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.interceptors.ClientContentEncodingAnnotationFeature is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.DefaultNumberWriter is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.SourceProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.FileProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.context.ContextFeature is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.interceptors.MessageSanitizerContainerResponseFilter is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.MultiValuedParamConverterProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.DocumentProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.DefaultBooleanWriter is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.ByteArrayProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.StringTextStar is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.StreamingOutputProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.ReaderProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.sse.SseEventSinkInterceptor is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.DataSourceProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,738 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.JaxrsFormProvider is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,739 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.FileRangeWriter is already registered.  2nd registration is being ignored.
2019-05-30 09:02:41,739 WARN  [org.jbo.res.res.i18n] (main) RESTEASY002155: Provider class org.jboss.resteasy.plugins.providers.InputStreamProvider is already registered.  2nd registration is being ignored.
```

**Comment by mkouba** [comment]
Steps to reproduce: once https://github.com/quarkusio/quarkus/pull/2635 is merged, just run `mvn clean test -Dtest=RestClientFallbackTest` in the `extensions/smallrye-rest-client/deployment` directory.

**Comment by gsmet** [comment]
@mkouba is it related to some new code? It's probably something we should fix before releasing 0.16.0, no?

**Comment by mkouba** [comment]
I don't believe it's new. I remember someone mentioned this on zulip.
